bika/lims/workflow/analysisrequest/events.py

```diff
@@ -92,6 +92,11 @@ def after_sample(obj):
     # is no need to transition neither the analyses nor partitions here
     _promote_transition(obj, 'sample')
 
+    # AutoReceiveSamples
+    auto = obj.bika_setup.getAutoReceiveSamples()
+    if auto and isTransitionAllowed(obj, 'receive'):
+        doActionFor(obj, 'receive')
```

Comment:
I tried that, then I realised that the guard must allow the transition always, so that the object may be manually received.  This is just the decision to invoke the transition.